### ISSUE TITLE
support for no_std

### ISSUE THREAD

**Issue by brenzi** [shepmaster/twox-hash#19]
Any chance you can support `#![no_std]`?

I'd like to use your crate within SGX, using https://github.com/baidu/rust-sgx-sdk

**Comment by shepmaster** [comment]
Could you try #19 and see if that's sufficient changes for your case?

**Comment by brenzi** [comment]
tested it:
```
twox-hash = { git ="https://github.com/shepmaster/twox-hash", branch="no-std", default-features = false }
```
and just adding this to my lib
```
extern crate twox_hash;
use twox_hash;
```

results in multiple  `error: duplicate lang item in crate `std`:`

**Comment by shepmaster** [comment]
> results in multiple `` error: duplicate lang item in crate `std`: ``

Do you have a way that I can reproduce the error?

**Comment by brenzi** [comment]
Yes, but I'm not sure it will build without installing Intel SGX SDK. Worth a try though.

This branch fails due to no_str trouble in twox-hash (revert the last commit and it should build):
https://github.com/scs/substraTEE-worker/tree/twox-test

```
cd enclave
cargo build --release
```
alternatively, try in the repo root
```
make
```

**Comment by shepmaster** [comment]
Oh, I think I know the problem — I didn't disable `std` support in the dependencies.

**Comment by shepmaster** [comment]
Would you mind checking the proposed branch again?

**Comment by brenzi** [comment]
great, works. thanks a lot. will you merge this into master?

**Comment by shepmaster** [comment]
> will you merge this into master

Yes, or something like it. I need to figure out how to create a useful automated test for that functionality to ensure that it doesn't regress.

**Comment by brenzi** [comment]
If I build your no-std branch with std support, I get:
```
error[E0433]: failed to resolve: maybe a missing `extern crate std_support;`?
   --> /home/abrenzikofer/.cargo/git/checkouts/twox-hash-d1ab4284274caa19/68108c8/src/thirty_two.rs:219:11
    |
219 | pub use ::std_support::thirty_two::RandomXxHashBuilder;
    |           ^^^^^^^^^^^ maybe a missing `extern crate std_support;`?
error[E0433]: failed to resolve: maybe a missing `extern crate std_support;`?
   --> /home/abrenzikofer/.cargo/git/checkouts/twox-hash-d1ab4284274caa19/68108c8/src/lib.rs:289:9
    |
289 | pub use std_support::sixty_four::RandomXxHashBuilder;
    |         ^^^^^^^^^^^ maybe a missing `extern crate std_support;`?
error[E0432]: unresolved import
  --> /home/abrenzikofer/.cargo/git/checkouts/twox-hash-d1ab4284274caa19/68108c8/src/lib.rs:46:9
   |
46 | pub use thirty_two::RandomXxHashBuilder as RandomXxHashBuilder32;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by brenzi** [comment]
your branch `no-std` currently fails to build with `std`:

```
error[E0433]: failed to resolve: maybe a missing `extern crate std_support;`?
   --> src/thirty_two.rs:219:11
    |
219 | pub use ::std_support::thirty_two::RandomXxHashBuilder;
    |           ^^^^^^^^^^^ maybe a missing `extern crate std_support;`?

error[E0433]: failed to resolve: maybe a missing `extern crate std_support;`?
   --> src/lib.rs:289:9
    |
289 | pub use std_support::sixty_four::RandomXxHashBuilder;
    |         ^^^^^^^^^^^ maybe a missing `extern crate std_support;`?

error[E0432]: unresolved import
  --> src/lib.rs:46:9
   |
46 | pub use thirty_two::RandomXxHashBuilder as RandomXxHashBuilder32;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

**Comment by brenzi** [comment]
cheers, fixed

**Comment by shepmaster** [comment]
Still figuring out how exactly to get a useful test in CI... :-)

**Comment by shepmaster** [comment]
Released as 1.3.0
